src/extension/rpc.ts

```diff
@@ -17,46 +17,34 @@ type RPCResponseMessageFormat =
   | { sourceId: number; result: unknown }
   | { sourceId: number; error: unknown };
 
-export type RPCMessage<
-  Name extends string,
-  Params extends RPCParams,
-  ReturnType,
-> = {
-  __name: Name;
-  __params: Params;
-  __returnType: ReturnType;
-};
+type MessageCollection = Record<string, (...parameters: [SafeAny]) => SafeAny>;
```

Comment:
Using `RPCParams` here unfortunately causes all kind of downstream problems, especially with the "empty" case of `Record<string, never>` - so I've gone for a `SafeAny` here. Is there a specific reason this has to be an object and not just any serializable value>

Reply:
No reason it had to be an object. This was one of the final improvements I hadn't come back to :). I like this better.